File: Adapter.java/src/com/company/EnemyTank.java
package com.company;
import java.util.Random;

public class EnemyTank implements EnemyAttacker{
    Random random = new Random();

    @Override
    public void shootBullet() {
        int attackDamage = random.nextInt(5) +1 ;
        System.out.println("Enemy Tank does " + attackDamage + " damages");
    }

    @Override
    public void driveForward() {
        int movement = random.nextInt(10) + 1;
        System.out.println("Enemy Tank moves " + movement + " spaces");
    }

    @Override
    public void assignDriver(String driverName) {
        System.out.println(driverName + " is driving the Tank");
    }
}

File: Adapter/EnemyTank.cs
namespace Adapter {
    internal class EnemyTank : IEnemyAttacker {
        private readonly Random random = new();

        public void ShootBullet() {
            int attackDamage = random.Next(5) + 1;
            Console.WriteLine("Enemy Tank does " + attackDamage + " damages");
        }

        public void DriveForward() {
            int movement = random.Next(10) + 1;
            Console.WriteLine("Enemy Tank moves " + movement + " spaces");
        }

        public void AssignDriver(string driverName) {
            Console.WriteLine(driverName + " is driving the Tank");
        }
    }
}
